### ISSUE TITLE
`NonEmptyObject<T>` type helper should have a constraint over an Object Type only for consistent result

### ISSUE THREAD

**Issue by ajitjha393** [ts-essentials/ts-essentials#335]
Hey @Beraliv,
Sorry for bothering you repeatedly :D
 

### Current Implementation for `NonEmptyObject<T>`  

```ts
type NonEmptyObject<T extends {}> = keyof T extends never ? never : T
```

Here as we can see there is a problem in the constraint that we have applied on the type parameter `T extends {}` 
which allows even the Primitive types to be passed on and hence the consistency of the type is not preserved.

### Problem

The above type instantiation should give an error and not resolve to a Primitive type Output

```ts

type T0 = NonEmptyObject<number> 
//   ^? number

type T1 = NonEmptyObject<string> 
//    ^? string


```

Code Playground - [https://tsplay.dev/WPpAkW](https://tsplay.dev/WPpAkW)

**Comment by ajitjha393** [comment]
###  Solution: 

```ts
type NonEmptyObject<T extends Record<PropertyKey, any>> = keyof T extends never ? never : T

```

Code Playground - [https://tsplay.dev/Nl2qRW](https://tsplay.dev/Nl2qRW)

**Comment by Beraliv** [comment]
It will be released in https://github.com/ts-essentials/ts-essentials/pull/337

**Comment by Beraliv** [comment]
Released in v9.3.1
